DynamicData/Cache/Internal/IndexCalculator.cs

```diff
@@ -67,26 +67,25 @@ public void Reset(ChangeAwareCache<TObject, TKey> cache)
             }
             else
             {
```

Comment:
This looks like a radical improvement to me.

Reply:
It is, but I'm still not sure if it won't actually break your stuff. I figured you've done the sorting the way you did, because there were important reasons, therefore my change may do something nasty, something that wouldn't be immediately obvious...

Reply:
I have been thinking about this. I remember now that the huge complexity with dd is there is no way to predict what a changeset may contain. For example there may be multiple updates to an object with the same key within a single changeset. This is particularly true when a batch operation is applied.  The proposed algorithm will work wonders but will fail in this case.

However there should be a simple solution to account for that case i.e check for duplicates within the same changeset. If there are no duplicates, apply the old algorithm, otherwise apply the new one.

Reply:
So in other words, should I check for uniqueness of objects inside the _list var?

Reply:
On second thoughts, I will take a fresh look at the code when I am on desktop later. I have been looking at this on my phone which is not ideal.

Reply:
I've just taken a fresh look at your code and I don't get one thing - it seems that except element[0], every other element "old index" is calculated the wrong way. Take a list of A, B and C letters that you want to sort in the inverse order. Boiling it down to a list of old positions, the result will contain information that C moved from 2, which is correct, and that B also moved from position 2 which doesn't seem to be correct. The question is then: should it be like that?

Reply:
I can honestly say I cannot remember why I did it that way. All I remember is sorting was the most complicated operator to get right in dd. Looking at the unit tests, they are rather sparse and require padding out. I suggest I do that first before making any more changes

Reply:
"For example there may be multiple updates to an object with the same key within a single changeset. This is particularly true when a batch operation is applied."

This is a problem I wanna address also when we can move to DD v6. I wanna conflate those changes if possible to keep only the most recent change to an item. Will be happy to write an operator for this when we get to v6